data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetGenesis.java

```diff
@@ -66,18 +81,42 @@ public GetGenesis(final DataProvider dataProvider, final JsonProvider jsonProvid
       })
   @Override
   public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
     final Optional<GenesisData> maybeData = getGenesisData();
     if (maybeData.isEmpty()) {
-      ctx.status(SC_NOT_FOUND);
+      request.respondWithCode(SC_NOT_FOUND);
       return;
     }
-    ctx.json(jsonProvider.objectToJSON(new GetGenesisResponse(maybeData.get())));
+    request.respondOk(maybeData.get());
   }
 
   private Optional<GenesisData> getGenesisData() {
     if (!chainDataProvider.isStoreAvailable()) {
       return Optional.empty();
     }
-    return Optional.of(chainDataProvider.getGenesisData());
+    return Optional.of(chainDataProvider.getStateGenesisData());
```

Comment:
Im just not sure if we need to go down this route of using a wrapper object... it feels like we shouldn't need it any more?

Reply:
It isn't a wrapper object anymore, that is getting the state `GenesisData` object